[vhostmd] add SAP virtualization metrics collection plugin

lmiccini opened the pull request:
Proposing a plugin to collect virtualization metrics exposed by the hypervisor.
Exposure of these metrics is one of the requirements for a virtualization platform to be certified by SAP.

Signed-off-by: Luca Miccini lmiccini@redhat.com

bmr-cymru commented:
* I'd suggest calling this "vhostmd" since that's the upstream project from which the `vm-dump-metrics` package builds. All those underscores make my eyes hurt..

* Need to test whether `virt-what` succeeds before using the result

* Not keen on the "vm-dump-metrics if it's there dd from a disk if not"; why not just always grab the dd and then decode it on the analysis host?

* You need a package or file check to control enablement of this plugin

lmiccini commented:
@bmr-cymru thanks:
- agree, fixed :)
- should be fixed in  b75407d
- the dd is kinda a "plan b". vm-dump-metrics should be *always* installed if SAP runs virtualized - it's in every SAP note/guide around virtualization (SAP support will ask to install it otherwise). 
- added a check for virt-what - can't think of a better solution at this moment :/
